src/context/mod.rs

```diff
@@ -335,6 +336,15 @@ pub trait Context<'a>: ContextInternal<'a> {
         let err = JsError::range_error(self, msg)?;
         self.throw(err)
     }
+
+    fn task<Perform, Complete, Output>(&mut self, perform: Perform) -> TaskBuilder<Self, Perform>
+    where
+        Perform: FnOnce() -> Complete + Send + 'static,
+        Complete: FnOnce(TaskContext) -> JsResult<Output> + Send + 'static,
+        Output: Value,
```

Comment:
FYI, the full set of bounds are required here because of this limitation: https://github.com/rust-lang/rust/issues/41078